src/Float.mo

```diff
@@ -1,71 +1,375 @@
-/// 64-bit Floating-point numbers
+/// Double precision (64-bit) floating-point numbers in IEEE 754 representation.
+///
+/// Common floating-point constants and functions.
+///
+/// Notation for special values in the documentation below:
+/// `+inf`: Positive infinity
+/// `-inf`: Negative infinity
+/// `nan`: "not a number" (can have different sign bit values, but `nan != nan` regardless of the sign).
+///
+/// Note:
+/// Floating point numbers have limited precision and operations may inherently result in numerical errors.
+///
+/// Examples of numerical errors:
+///   ```motoko name=initialize
+///   assert(0.1 + 0.1 + 0.1 == 0.3); // Fails!
+///   ```
+///
+///   ```motoko name=initialize
+////  assert(1e16 + 1.0 != 1e16); // Fails!
+///   ```
+///
+////  (and many more cases)
+///
+/// Advice:
+/// * Floating point number comparisons by `==` or `!=` are discouraged. Instead, it is better to compare
+///   floating-point numbers with a numerical epsilon.
+///
+///   Example:
+///   ```motoko name=initialize
+///   import Float "mo:base/Float";
+///   let epsilon = 1e-6; // This depends on the application case (needs a numerical error analysis).
+///   let equals = Float.abs(x - y) < epsilon;
+///   ```
+///
+/// * For absolute precision, it is recommened to encode the fraction number as a pair of a Nat for the base
+//    and a Nat for the exponent (decimal point).
+///
 
 import Prim "mo:⛔";
 import Int "Int";
 
 module {
 
-  /// 64-bit floating point numbers.
+  /// 64-bit floating point number type.
   public type Float = Prim.Types.Float;
 
   /// Ratio of the circumference of a circle to its diameter.
+  /// Note: Limited precision.
   public let pi : Float = 3.14159265358979323846; // taken from musl math.h
 
   /// Base of the natural logarithm.
+  /// Note: Limited precision.
   public let e : Float = 2.7182818284590452354; // taken from musl math.h
 
   /// Returns the absolute value of `x`.
+  ///
+  /// Special cases:
+  /// ```
+  /// | Argument `x` | Result `abs(x)` |
+  /// | ------------ | --------------- |
+  /// | `+inf`       | `+inf`          |
+  /// | `-inf`       | `+inf`          |
+  /// | `nan`        | `nan`           |
+  /// | `-0.0`       | `0.0`           |
+  /// ```
+  ///
+  ///   Example:
+  ///   ```motoko name=initialize
```

Comment:
This example fails when ran on the webpage. I think you have to import the `Float` module either in this snippet, or in another snippet that is included here using `name` and `import`.

E.g.
Snippet being named[
](https://github.com/dfinity/motoko-base/blob/707ad38d5da578907719f154c6278aba9b0b768b/src/Buffer.mo#L27)
[Snippet being included](https://github.com/dfinity/motoko-base/blob/707ad38d5da578907719f154c6278aba9b0b768b/src/Buffer.mo#L74)

Reply:
Thanks a lot. I have (hopefully) fixed this in meantime.